plugins/push/frontend/public/javascripts/countly.models.js

```diff
@@ -347,13 +414,32 @@
                 dataType: "json"
             });
         },
+        fetchMediaMetadata: function(url) {
+            var self = this;
+            return new Promise(function(resolve, reject) {
```

Comment:
You didn't need to wrap the CV.$.ajax with Promise, as it is already done in the CV.$.ajax itself. You might use then/catch instead of success/error, as well.

Reply:
Yes, that's what I have for cases when I do not map the data to model. Unfortunately, I will have to keep them at least I am afraid for the upcoming release.